File: python-course/com/example/selenium/spree/pages/BasePage.py
class BasePage:
    def __init__(self, test, title, url):
        self.test = test
        self.title = title
        self.url = url

    def validateTitle(self):
        title = self.test.webDriver.title
        self.test.assertEqual(self.title, title)

    def validateUrl(self):
        url = self.test.webDriver.current_url
        self.test.assertEqual(self.url, url)

File: java-course/src/test/java/com/example/selenium/spree/pages/BasePage.java
package com.example.selenium.spree.pages;

import org.openqa.selenium.remote.RemoteWebDriver;
import org.testng.Assert;

public class BasePage {
    protected RemoteWebDriver webDriver;
    private String title;
    private String url;

    public BasePage(RemoteWebDriver webDriver, String title, String url) {
        this.webDriver = webDriver;
        this.title = title;
        this.url = url;
    }

    public void validateTitle() {
        String title = webDriver.getTitle();
        Assert.assertEquals(title, this.title);
    }

    public void validateUrl() {
        String url = webDriver.getCurrentUrl();
        Assert.assertEquals(url, this.url);
    }
}
